src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/RefreshableRegistry.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.registry;
+
+public interface RefreshableRegistry {
```

Comment:
Consider using composition here rather than implementing an interface. For example, imagine constructing the registries with a cache refresher object. The cache itself should know when things need to be refreshed. This interface seems a bit rigid, in the sense that the registry is opaque. The thing responsible for driving the refresh can't really do anything intelligent.